lib/rubygems/ext/ext_conf_builder.rb

```diff
@@ -50,11 +50,16 @@ def self.build(extension, dest_path, results, args=[], lib_dir=nil, extension_di
   end
 
   def self.get_relative_path(path, base)
-    path.split("/").zip(base.split("/")).inject(String.new) do |result, (path_component, base_component)|
-      next result if path_component == base_component
-      result.prepend("../") if base_component
-      result.concat("#{path_component}/") if path_component
-      result
+    absolute_path = File.expand_path(path)
+    absolute_base = File.expand_path(base)
+
+    backtracks = 0
+    prefix = absolute_base
+    until absolute_path.start_with?(prefix)
+      prefix = File.dirname(prefix)
```

Comment:
Can you add a raise in here if prefix becomes "." so that it doesn't construct a wrong path or loop forever if this, somehow, gets called with paths that don't have a common ancestor?

Maybe overly paranoid since the original was no better, but looking at [Pathname's implementation](https://docs.ruby-lang.org/en/master/Pathname.html#method-i-relative_path_from) maybe it doesn't hurt to exit explicitly in that case?

Reply:
It seems strange that `File.dirname` ever returns `.`, but it does return that on Windows when reaching the root drive, not sure why. If `base` and `path` were on different drives, then we could potentially end up looping forever, indeed? That said, I'm not sure  `base` and `path` can ever be on different drives?

Reply:
Fun! I thought the calls to `expand_path` would take care of everything, but I guess not 😅.

Reply:
Pushed up a change to raise if `File.dirname` doesn't change anything. That would mean we've gone as far as we can go and didn't find a match. I think that covers both the "." case and the different drives case. 🤞🏻